src/Compilers/VisualBasic/Portable/Parser/ParseDeclarationStatement.vb

```diff
@@ -0,0 +1,365 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+'-----------------------------------------------------------------------------
+' Contains the definition of the Scanner, which produces tokens from text 
+'-----------------------------------------------------------------------------
+
+Imports System.Runtime.InteropServices
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.PooledObjects
+Imports Microsoft.CodeAnalysis.Syntax.InternalSyntax
+Imports Microsoft.CodeAnalysis.Text
+Imports CoreInternalSyntax = Microsoft.CodeAnalysis.Syntax.InternalSyntax
+Imports InternalSyntaxFactory = Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.SyntaxFactory
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
```

Comment:
why was all this code created?

Reply:
I separated out some of the parser methods into their own files. I also moved, some the of the xml parser  from `Parser.vb` is not in the `ParseXML.vb` file.